src/igvfd/report.py

```diff
@@ -56,20 +62,37 @@ def _convert_camel_to_snake(type_str):
 @view_config(route_name='report_download', request_method='GET')
 def report_download(context, request):
     downloadtime = datetime.datetime.now()
-
-    types = request.params.getall('type')
-    if len(types) != 1:
-        msg = 'Report view requires specifying a single type.'
-        raise HTTPBadRequest(explanation=msg)
-
+    request.GET['limit'] = '0'
+    fr = FieldedResponse(
+        _meta={
+            'params_parser': ParamsParser(request)
+        },
+        response_fields=[
+            MultipleTypesReportWithFacetsResponseField(
+                default_item_types=DEFAULT_ITEM_TYPES,
+                reserved_keys=RESERVED_KEYS,
+            ),
+            ColumnsResponseField(),
+
+        ]
+    )
+    response = fr.render()
+    facets = response.json['facets']
+    abstract_types = get_abstract_types(request)
+    types_in_search_result = []
+    for facet in facets:
+        if facet['field'] == 'type':
+            for term in facet['terms']:
+                type_name = term['key']
+                if type_name not in abstract_types:
+                    types_in_search_result.append(term['key'])
+            break
```

Comment:
What is the `break` doing?

Reply:
break after find facet "type" and get all the term keys.